crates/wasi-common/yanix/src/lib.rs

```diff
@@ -68,3 +68,19 @@ macro_rules! impl_is_minus_one {
 }
 
 impl_is_minus_one! { i32 i64 isize }
+
+/// Convert an `AsRef<Path>` into a `CString`.
+fn cstr<P: AsRef<std::path::Path>>(path: P) -> std::io::Result<std::ffi::CString> {
```

Comment:
We already pull in `std::io::{Error, Result}` into scope in this module, so we might as well use it here. Also, while we already do that, I guess we could pull in `std::path::Path` and `std::ffi::CString`. It's a minor stylistic nit, so feel free to ignore!

Reply:
Fixed.